File: tests/unit/extract-plain.test.ts
import { describe, expect, it } from 'vitest';
import { parseEmailText } from '../../src/pipeline/extract/plainText.js';

describe('parseEmailText', () => {
  it('extracts basic rows', () => {
    const text = `\nВВГнг 3х2.5 100 шт\nКабель NYM 10 м\n`;
    const items = parseEmailText(text);
    expect(items.length).toBe(2);
    expect(items[0].qty).toBe(100);
    expect(items[1].qty).toBe(10);
  });

  it('skips signature rows', () => {
    const text = `С уважением\nИван\nКабель 5 шт`;
    const items = parseEmailText(text);
    expect(items.length).toBe(1);
  });

  it('supports mixed separators', () => {
    const text = `Позиция: Провод ПВС; кол-во 1,5 м`;
    const items = parseEmailText(text);
    expect(items[0].qty).toBe(1.5);
  });

  it('ignores tiny non-product line', () => {
    const text = `ok\nAB\nКабель 2 шт`;
    const items = parseEmailText(text);
    expect(items.length).toBe(1);
  });

  it('handles code-like line', () => {
    const text = `ELC0100203802 3 шт`;
    const items = parseEmailText(text);
    expect(items.length).toBe(1);
    expect(items[0].nameOrCode).toContain('ELC0100203802');
  });
});

File: internal/pipeline/extract_plain_test.go
package pipeline

import "testing"

func TestParseEmailText(t *testing.T) {
	text := "\nВВГнг 3х2.5 100 шт\nКабель NYM 10 м\n"
	items := parseEmailText(text)
	if len(items) != 2 {
		t.Fatalf("len=%d", len(items))
	}
	if items[0].Qty == nil || *items[0].Qty != 100 {
		t.Fatalf("qty1 bad")
	}
	if items[1].Qty == nil || *items[1].Qty != 10 {
		t.Fatalf("qty2 bad")
	}
}

func TestParseEmailTextCodeLike(t *testing.T) {
	items := parseEmailText("ELC0100203802 3 шт")
	if len(items) != 1 {
		t.Fatalf("len=%d", len(items))
	}
	if items[0].NameOrCode == nil || *items[0].NameOrCode != "ELC0100203802" {
		t.Fatalf("nameOrCode=%v", items[0].NameOrCode)
	}
}
